### PR TITLE
[MM-50746] - Edits to messages persist after cancelling

### LINKED ISSUE #22372: Edits to messages persist after cancelling
#### Summary
When you cancel edits to a message and try to edit it again, your previous changes persist and the original message is gone. 

#### Steps to reproduce
1. Open mattermost (version 7.8.0)
2. Find any message that was posted by your account (I could do this both with messages in channels and in my DMs with myself).
3. Edit the message such that the content is different than the original message.
4. Cancel your edits to the message.
5. Hit the edit button again. Note that the original message does not appear in the text box. 

#### Expected behavior
When I hit edit the second time, the original message should reappear, not my previously cancelled edits. 

#### Observed behavior (that appears unintentional) 
When I hit edit the second time, the edited message that I previously cancelled is still showing in the text area.

#### Possible fixes
I assume this is, in React terms, a state clearing issue. Rather than setting the edited message back to the original message `onEditCancel`, the edited value remains unchanged and persists such that you never reset to the original message text.

### PR DESCRIPTION
**Pull request by gitstart**
### Summary
Edits to messages persist after canceling

### Demo video
https://www.loom.com/share/330e0189b6a1445a9cb7652401363280

#### Release note
```release-note
Discards edits to messages after canceling
```

### CODE REVIEW

**Comment by gitstart** [comment]
@hmhealey you can have a look at this. Thanks. cc @M-ZubairAhmed

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for the PR! This looks pretty good to me, but could you add a unit tests or E2E test for this behaviour as well?

Unit tests are easier to run, but since this component doesn't have any yet, they might be trickier to set up. There's quite a few E2E tests for this component already though such as the ones in `e2e/cypress/tests/integration/messaging/edit_message_spec.js`. Either of those would work fine though

**Comment by gitstart** [comment]
@hmhealey an e2e test has been added. I have a question. Where do I get these ids added to e2e tests e.g `MM-T121` in  `it('MM-T121 Escape should not.... `.

File: e2e/cypress/tests/integration/messaging/edit_message_spec.js [hunk 1]
```diff
@@ -145,4 +145,42 @@ describe('Edit Message', () => {
             cy.get(postText).should('have.text', `${secondMessage} Another new message Edited`);
         });
     });
+
+    it('should discard any changes made after cancelling the edit and opening the edit textbox again should display the original message', () => {
```
Comment (AshishDhama) [inline e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: @furqanmlk cloud you please help with the test id here

**Comment by hmhealey** [comment]
> @hmhealey an e2e test has been added. I have a question. Where do I get these ids added to e2e tests e.g `MM-T121` in `it('MM-T121 Escape should not.... `.

Our QA team usually handles adding those. I don't think it's mandatory for all E2E tests to have those IDs though

**Review by hmhealey — APPROVED** [review]
Thanks for adding that test! I checked it out locally and it looks good to me

**Comment by M-ZubairAhmed** [comment]
@gitstart the new e2e should be first added in https://github.com/mattermost/mattermost-test-management the new ID will be assigned over there, then the e2e with the new id should be added here. Please note that in the above repo the tests cases should be written in simple english.

File: e2e/cypress/tests/integration/messaging/edit_message_spec.js (hunk 1 above)
Comment (AshishDhama) [reply e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: @furqanmlk is it no required

Comment (furqanmlk) [reply e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: Hey @AshishDhama, Sorry for the late response, In order to get the `test ID`, you need to create a PR in test management repo `https://github.com/mattermost/mattermost-test-management`.
For example https://github.com/mattermost/mattermost-test-management/pull/38

**Comment by gitstart** [comment]
@M-ZubairAhmed  the test PR has been created here https://github.com/mattermost/mattermost-test-management/pull/83. You can have a look at it. Thanks. cc @furqanmlk , @AshishDhama

File: e2e/cypress/tests/integration/messaging/edit_message_spec.js (hunk 1 above)
Comment (M-ZubairAhmed) [reply e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: @gitstart can you add the test id here

Comment (M-ZubairAhmed) [inline e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: Can you get the test id from https://github.com/mattermost/mattermost-test-management/pull/83

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: e2e/cypress/tests/integration/messaging/edit_message_spec.js (hunk 1 above)
Comment (M-ZubairAhmed) [reply e2e/cypress/tests/integration/messaging/edit_message_spec.js:149]: It will be provided in that PR soon.

**Comment by M-ZubairAhmed** [comment]
Hey there!, Thank you for your PR.
Webapp is now part of mattermost monorepo now at https://github.com/mattermost/mattermost-server/tree/master/webapp Would you be interested in opening the PR over there?

**Comment by hmhealey** [comment]
The web app now lives in the `webapp/channels` over in https://github.com/mattermost/mattermost-server. We decided to move it to be a monorepo to make both development and our CI pipelines easier, but it's unfortunately disruptive right now. We've updated [our setup docs](https://developers.mattermost.com/contribute/developer-setup/) and added [some steps for migrating to working with the monorepo](https://developers.mattermost.com/contribute/monorepo-migration-notes/).

Copying the changes over there hopefully shouldn't be too hard. There's a few changes to file paths (such as the code living in a dedicated `src` subdirectory), but most things should look the same as they do now. 

Let us know if you could use any help with this. Hopefully the process is smooth.

**Comment by gitstart** [comment]
@hmhealey This PR has been moved to the monorepo https://github.com/mattermost/mattermost-server/pull/22676. cc @M-ZubairAhmed

**Pull request closed without merging**